src/audio/selector/selector.c

```diff
@@ -579,7 +579,7 @@ static void build_config(struct comp_data *cd, struct module_config *cfg)
 	enum sof_ipc_frame __sparse_cache frame_fmt, valid_fmt;
 	int i;
 
-	frame_fmt = cfg->base_cfg.audio_fmt.depth;
+	frame_fmt = (int)cfg->base_cfg.audio_fmt.depth;
```

Comment:
Well, when `sof/lib/dai.h` is included, `depth` is a `uint32_t` type member of `dai_plat_fifo_data` struct.
However when we compile with IPC4 config option,  header `src\include\ipc4\base-config.h` is uder and `depth` is `enum ipc4_bit_depth` type what causes the warning:

> implicit conversion from enumeration type 'enum ipc4_bit_depth' to different enumeration type 'enum sof_ipc_frame' [-Wenum-conversion]

Enums are implicitly cast to integral types but not to other enums. But thanks for the catch since I should cast to `(uint32_t)` here to avoid value overflow.

Reply:
> However when we compile with IPC4 config option depth is enum ipc4_bit_depth instead of uint32_t

That definitely deserves a (short) comment IMHO.

Reply:
Done in https://github.com/thesofproject/sof/pull/7096 .